portal/src/app/components/home/components/experiment/components/table-options/components/select-columns-popup/SelectColumnsPopup.tsx

```diff
@@ -0,0 +1,84 @@
+import { useCallback, useState } from 'react';
+import { AttSelectOption } from '../../../../../../../../shared/components/att-select';
+import styles from './SelectColumnsPopup.module.scss';
+import { SELECT_COLUMNS_EN } from './translate/en';
+import CloseSvg from '../../../../../../../../../assets/images/close.svg';
+import CheckedSvg from '../../../../../../../../../assets/images/checked.svg';
+import UnCheckedSvg from '../../../../../../../../../assets/images/unchecked.svg';
+import { Button, ButtonActionType, ButtonSize, ButtonStyleType } from '../../../../../../../../shared/components/att-button';
+
+const CloseAriaLabel: string = 'Close';
+
+export interface SelectColumnsPopupProps {
+  data: AttSelectOption[];
+  onPopupClose: () => void;
+  onColumnsSelected: (selectedColumns: AttSelectOption[]) => void;
```

Comment:
I think you can use just one event , like `onPopupClose` , and when you send ` onPopupClose: (selectedColumns?: AttSelectOption[]) => void;` with `selectedColumns` that said you clicked on save button else (without data) that said you click on close button

thats say, that you can remove `onColumnsSelected` event and use only `onPopupClose` event for both options (save/close)

Reply:
please use one of the event instead both, like my example : onPopupClose: (selectedColumns?: AttSelectOption[]) => void;

Reply:
I was think about making that change and decided to leave it that way for two reasons:
1. Separate this into two events make it more understandable since it is totally two different things.
2. Implement it demands a little refactoring. Making those slight changes will not make the events work as expected.